eth/handler.go

```diff
@@ -112,7 +112,7 @@ func NewProtocolManager(config *core.ChainConfig, singleMiner bool, networkId in
 		quitSync:    make(chan struct{}),
 		raftMode:    raftMode,
 	}
-	if singleMiner {
+	if assumeSyncedInitially {
 		manager.synced = uint32(1)
```

Comment:
@hermanjunge have you experienced the reported issue with this PR? I have multiple tests sending transactions across multiple nodes with different roles and I didn't see a TX propagation issue.